tests/common/tasks/test_task_manager.py

```diff
@@ -130,3 +69,137 @@ async def test_weekly_cadence_with_last_run():
+            await self.bot.wait_until_ready()
+
+    TOKEN = BotConstants.token
+    CHANNEL = Tests.test_channel
+    bot = get_bot()
+    bot.add_cog(TestCog(bot, CHANNEL))
+    bot.run(TOKEN)
+
+
+TOKEN = BotConstants.token
+CHANNEL = Tests.test_channel
+# TODO: extract dependency and mock creation into a fixture
+
+
+@pytest.mark.discord_bot
+def test_reporting():
+    """
+    Tests the full BotTasks cog (currently only consisting of the reporting task)
+    with injected cache, cadence, settings
+    """
+    loop_settings = {
+        "task_wakeup_period_minutes": 1,
+        "enable": True,
+    }
+    bot = get_bot()
+
+    # the reporting task should run every minute, since the cadence
+    # will always return "it's time to run"
+    bot.add_cog(ReportingTask(bot, MockCache(), MockCadence(None), loop_settings))
+    bot.run(TOKEN)
+
+
+@pytest.mark.discord_bot
+def test_reporting_with_cache():
+    loop_settings = {
+        "task_wakeup_period_minutes": 1,
+        "enable": True,
+    }
+
+    mock_cache = MockCache()
+    cadence = MockCadence(None)
+    bot = get_bot()
+
+    # set cache entry to mock as if the bot had sent the report,
+    # crashed, and is now re-running after restart to ensure the
+    # report is not sent constantly if the bot is crashing
+    asyncio.run_coroutine_threadsafe(
+        mock_cache.set(
+            ReportingTask.REPORT_LAST_SENT_DATETIME_CACHE_KEY,
+            datetime.now().strftime(DATETIME_CACHE_FMT),
+        ),
+        bot.loop,
+    )
+
+    bot.add_cog(
+        ReportingTask(
+            bot, mock_cache, cadence, loop_settings, reporting_channel=CHANNEL
+        )
+    )
+    # TODO: assert that the task is not run
+    bot.run(TOKEN)
+
+
+@pytest.mark.discord_bot
+def test_task_disable():
+    loop_settings = {
+        "enable": False,
+    }
+
+    mock_cache = MockCache()
+    cadence = MockCadence(None)
+    bot = get_bot()
+
+    bot.add_cog(
+        ReportingTask(
+            bot, mock_cache, cadence, loop_settings, reporting_channel=CHANNEL
+        )
+    )
+    # TODO: assert that the task does not run
+    bot.run(TOKEN)
+
+
+@pytest.mark.discord_bot
+def test_task_hourly():
+    loop_settings = {
+        "task_wakeup_period_minutes": 10,
+        "enable": True,
+    }
+
+    mock_cache = MockCache()
+    cadence = MockCadence(None)
+    bot = get_bot()
+
+    # set cache entry to mock as if the bot had sent the report,
+    # crashed, and is now re-running after restart to ensure the
+    # report is not sent constantly if the bot is crashing
+    asyncio.run_coroutine_threadsafe(
+        mock_cache.set(
+            ReportingTask.REPORT_LAST_SENT_DATETIME_CACHE_KEY,
+            datetime.now().strftime(DATETIME_CACHE_FMT),
+        ),
+        bot.loop,
+    )
+
+    bot.add_cog(
+        ReportingTask(
+            bot, mock_cache, cadence, loop_settings, reporting_channel=CHANNEL
+        )
+    )
+    # TODO: assert that the task is not run
+    bot.run(TOKEN)
```

Comment:
This should probably be added if it isn't too much work

Reply:
these are manual integration tests that depend on discord + the db being set up. once unit tests are up for this we can add some assertions so they can run on PRs.